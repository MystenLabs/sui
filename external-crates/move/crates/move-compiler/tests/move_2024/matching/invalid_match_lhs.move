module 0x42::m;

public enum Maybe<T> {
    Just(T),
    Nothing
}

fun test(z: &mut Maybe<u64>) {
    let { match (z) { Maybe::Just(n) => n, Maybe::Nothing => abort 0 } } = 5;
}


