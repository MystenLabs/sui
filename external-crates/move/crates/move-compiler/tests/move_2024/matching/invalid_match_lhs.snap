warning[W09002]: unused variable
  ┌─ tests/move_2024/matching/invalid_match_lhs.move:8:10
  │
8 │ fun test(z: &mut Maybe<u64>) {
  │          ^ Unused parameter 'z'. Consider removing or prefixing with an underscore: '_z'
  │
  = This warning can be suppressed with '#[allow(unused_variable)]' applied to the 'module' or module member ('const', 'fun', or 'struct')

error[E01002]: unexpected token
  ┌─ tests/move_2024/matching/invalid_match_lhs.move:9:9
  │
9 │     let { match (z) { Maybe::Just(n) => n, Maybe::Nothing => abort 0 } } = 5;
  │         ^
  │         │
  │         Unexpected '{'
  │         Expected a variable or struct name

